Controller/RegistrationController.php

```diff
@@ -60,8 +60,15 @@ public function registerAction(Request $request)
 
             $userManager->updateUser($user);
 
-            if (null === $response = $event->getResponse()) {
-                $url = $this->generateUrl('fos_user_registration_confirmed');
+            $confirmationEnabled = $this->container->getParameter('fos_user.registration.confirmation.enabled');
+            if($confirmationEnabled){
+                if (null === $response = $event->getResponse()) {
+                    $url = $this->generateUrl('fos_user_registration_confirmed');
+                    $response = new RedirectResponse($url);
+                }
+            }
+            else {
```

Comment:
Maybe move this line: https://github.com/FriendsOfSymfony/FOSUserBundle/pull/2152/files#diff-1e84feefcfa63cdf2badad975a6cb5aaR75

Before the if statement and return directly the response to avoid the `else` case.

cc @stof @XWB